src/org/zaproxy/zap/extension/globalexcludeurl/DialogAddToken.java

```diff
@@ -117,6 +123,18 @@ protected void init() {
         getEnabledCheckBox().setSelected(true);
         getDescTextField().setText("");
         token = null;
+        
+		// Handle escape key to close the dialog
+		KeyStroke escape = KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0, false);
+		AbstractAction escapeAction = new AbstractAction() {
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				dispatchEvent(new WindowEvent(DialogAddToken.this, WindowEvent.WINDOW_CLOSING));
```

Comment:
It should call `clearAndHide` (or `clearFields` and `dispose`) instead (same behaviour as Cancel button).